core/spec/lib/spree/core/controller_helpers/order_spec.rb

```diff
@@ -84,6 +84,19 @@
         controller.associate_user
       end
     end
+
+    context "configuration associate_user_in_authentication_extension is false" do
+      before do
+        stub_spree_preferences(associate_user_in_authentication_extension: false)
+      end
```

Comment:
I think we should move the above specs into this context since they will start failing once we update the version and also they are only relevant in the case where the value of this setting is `false`.

We could also add a test for the new behaviour which should not associate the user nor show a deprecation warning.

Reply:
> We could also add a test for the new behaviour which should not associate the user nor show a deprecation warning.

Agree

> I think we should move the above specs into this context since they will start failing once we update the version and also they are only relevant in the case where the value of this setting is false.

What I've been doing is still to test that the deprecation warning is shown. As it's useful for users, I like to have it covered. We'd remove it alongside the setting on `v4.0`. But, in this case, mine is not a strong opinion, so I'd be good with any outcome.